### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by kevincris**

### CODE REVIEW

File: src/continents.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+'use strict'
```
Comment (ppetzold) [inline src/continents.ts]: remove

File: src/continents.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { regions } from './regions.js'
+
+export interface Continent {
+    name: string
+    regions: string[]
+    countries: []
+}
+
+export interface Continents {
+    asia: Continent | {}
```
Comment (ppetzold) [inline src/continents.ts]: remove ` | {}` everywhere

and then just cast below e.g. `asia: {} as any,`

File: src/country_csv_to_json.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+'use strict'
```
Comment (ppetzold) [inline src/country_csv_to_json.ts]: remove

File: src/country_csv_to_json.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import csv from 'csv'
+import canonicalJSON from 'canonical-json'
+import fs from 'fs'
+
+let output = []
+const countriesFilename = 'countries.csv'
+const deletedCountriesFilename = 'deleted_countries.csv'
+
+function readFile(filename) {
+    return new Promise(function (resolve) {
```
Comment (ppetzold) [inline src/country_csv_to_json.ts]: use arrow functions everywhere

File: src/currency_csv_to_json.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+'use strict'
```
Comment (ppetzold) [inline src/currency_csv_to_json.ts]: remove

File: src/currency_csv_to_json.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import fs from 'fs'
+
+let output = []
+// read in the CSV
+const csvFile = path.join(__dirname, 'currencies.csv')
+const input = fs.createReadStream(csvFile)
+
+const parser = csv.parse({ columns: true })
+
+parser.on('readable', function () {
```
Comment (ppetzold) [inline src/currency_csv_to_json.ts]: same

File: src/language_csv_to_json.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
+'use strict'
```
Comment (ppetzold) [inline src/language_csv_to_json.ts]: remove

File: src/normalize_csv.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
+'use strict'
```
Comment (ppetzold) [inline src/normalize_csv.ts]: remove

File: src/normalize_csv.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+const firstHeader = process.argv[2]
+
+let output = []
+
+// read in the CSV
+const input = process.stdin
+
+const parser = csv.parse({ columns: true })
+
+parser.on('readable', function () {
```
Comment (ppetzold) [inline src/normalize_csv.ts]: same

File: src/language_csv_to_json.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+let output = []
+
+// read in the CSV
+const csvFile = path.join(__dirname, 'languages.csv')
+const input = fs.createReadStream(csvFile)
+
+const parser = csv.parse({ columns: true })
+
+parser.on('readable', function () {
```
Comment (ppetzold) [inline src/language_csv_to_json.ts]: same

File: src/regions.ts [hunk 11]
```diff
@@ -0,0 +1,437 @@
+'use strict'
```
Comment (ppetzold) [inline src/regions.ts]: remove

File: src/regions.ts [hunk 12]
```diff
@@ -0,0 +1,437 @@
...
+
+export const regions: any = {
```
Comment (ppetzold) [inline src/regions.ts]:
```typescript
interface Region {
  name: string
  countries: string[]
}
```

**Review by ppetzold — CHANGES_REQUESTED** [review]

File: src/continents.ts (hunk 1 above)
Comment (kevincris) [reply src/continents.ts]: ok sir

File: src/continents.ts (hunk 2 above)
Comment (kevincris) [reply src/continents.ts]: yes sir

File: src/country_csv_to_json.ts (hunk 4 above)
Comment (kevincris) [reply src/country_csv_to_json.ts]: ok sir

**Review by ppetzold — APPROVED** [review]

**Pull request merged**